ArchiSteamFarm/Bot.cs

```diff
@@ -2836,6 +2837,136 @@ public sealed class Bot : IAsyncDisposable {
 			}
 		}
 
+		private void OnInventoryChanged() {
+			Utilities.InBackground(CardsFarmer.OnNewItemsNotification);
+
+			if (BotConfig.BotBehaviour.HasFlag(BotConfig.EBotBehaviour.DismissInventoryNotifications)) {
+				Utilities.InBackground(ArchiWebHandler.MarkInventory);
+			}
+
+			if (BotConfig.SendSetsOnCompleted) {
+				Utilities.InBackground(SendCompletedSets);
+			}
+		}
+
+		private async Task SendCompletedSets() {
+			IDocument? badgePage = await ArchiWebHandler.GetBadgePage(1).ConfigureAwait(false);
+
+			if (badgePage == null) {
+				ArchiLogger.LogGenericWarning(Strings.WarningCouldNotCheckBadges);
+
+				return;
+			}
+
+			List<IElement> craftButtons = badgePage.SelectNodes("//a[@class='badge_craft_button']");
+
+			if (craftButtons.Count == 0) {
+				// no craftable badges
+				return;
+			}
+
+			HashSet<ulong> appIds = new HashSet<ulong>(craftButtons.Count);
+
+			foreach (string? badgeUri in craftButtons.Select(htmlNode => htmlNode.GetAttribute("href"))) {
+				if (string.IsNullOrEmpty(badgeUri)) {
+					ArchiLogger.LogNullError(nameof(badgeUri));
+
+					continue;
+				}
+
+				const string uriSegmentBeforeAppId = "/gamecards/";
+				int index = badgeUri.LastIndexOf(uriSegmentBeforeAppId, StringComparison.Ordinal);
+
+				if (index < 0) {
+					ArchiLogger.LogNullError(nameof(index));
+
+					continue;
+				}
+
+				index += uriSegmentBeforeAppId.Length;
+
+				if (index >= badgeUri.Length) {
+					ArchiLogger.LogNullError(nameof(index));
+
+					continue;
+				}
+
+				string? appIdText = string.Concat(badgeUri.Skip(index).TakeWhile(char.IsDigit));
```

Comment:
This is just awful, check how I extract appID from the same place in `CardsFarmer` and use logic there, probably you could extract it into its own function.

Reply:
In CardsFarmer the only call to `uint.TryParse(...)` is after splitting some attribute value on `_`. Can't do that here, as the selected element only has the `href` attribute pointing to the badge page directly. I know this isn't beautiful tho and am open for improvements on this one

Reply:
Ok, why you can't split on `/` and take last entry?

Reply:
That will probably work. Let me check

Reply:
It's the same approach, just because we split on `_` there and `/` here is pretty misc - doesn't justify general function for parsing that though, so no need to merge those parts.